conans/test/integration/command/install/install_test.py

```diff
@@ -459,6 +459,28 @@ def test_create_cli_override(self, client):
         client.run("create . pkg/0.1@ --require-override=zlib/2.0")
         assert "zlib/2.0: Already installed" in client.out
 
+    def test_conditional(self, client):
+        # https://github.com/conan-io/conan/issues/10045
+        client.save({"conanfile.py": GenConanfile()})
+        client.run("create . cuda/10.2@")
+        client.run("create . cuda/10.3@")
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile
+            class Pkg(ConanFile):
+                options = {"with_cuda":[True,False]}
+                default_options = {"with_cuda":True}
+                def requirements(self):
+                    if self.options.with_cuda:
+                        self.requires('cuda/[>=10.1 <=11.4.2]')
+            """)
+        client.save({"conanfile.py": conanfile})
+        client.run("create . opencv/1.0@ --require-override=cuda/10.2")
+        assert "cuda/10.2" in client.out
+        assert "cuda/10.3" not in client.out
+        client.run("install . --require-override=cuda/10.2")
+        assert "cuda/10.2" in client.out
+        assert "cuda/10.3" not in client.out
```

Comment:
Should we check that `create opencv/1.0@` brings `cuda/10.3`?

Reply:
Done